independent-projects/bootstrap/classloader-commons/src/main/java/io/quarkus/commons/classloading/ClassloadHelper.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.commons.classloading;
+
+public final class ClassloadHelper {
+
+    private ClassloadHelper() {
+        //Not meant to be instantiated
+    }
+
+    /**
+     * Sanitizes the string passed as argument, by removing any "/"
+     * symbol at the end or at the beginning.
+     *
+     * @param name the string to be sanitized
+     * @return a new String with the offending characters removed, or potentially the same String
+     *         if no modifications were required.
+     */
+    public static String sanitizeName(final String name) {
```

Comment:
I'm not sure that sanitization - as a concept - belongs here (or anywhere). The idea of taking a "dirty" name and processing it into a "clean" name means that at some point, we don't know where names are coming from, or why.

For example, are we translating a URI path to a class name? If so, then maybe `pathToClassInternalName` is a better name (and behavioral contract). Then things which produce a URI can deterministically turn those things into class names, and the question of "do I need to sanitize this name?" no longer exists.

Are we determining the relative URI path of a (possibly absolute) resource path? Again, a targeted method with a clear name would be preferable.

Reply:
I wonder if the methods in this class shouldn't be in https://github.com/smallrye/smallrye-common/blob/main/classloader/src/main/java/io/smallrye/common/classloader/ClassPathUtils.java instead

Reply:
Good question. I'd prefer them to be in Quarkus as I expect the set of things we need exactly to change quite often..  for example take David's point he's already suggesting we should get rid of this method, and I agree.. but I like doing things in small incremental steps.

Moving things in a different project would really slow us down; personally I'd not work on further improvements as I can only afford some occasional time for such things, not making it a multi-month project.

Reply:
My only concern is that this is public API.

Reply:
> My only concern is that this is public API.

ah.. I agree that's horrible. I'm not sure what to do then, I was aiming at some form of private module but that's not the case.

I'm tempted to close the PR until we have some proper methods to expose. `pathToClassInternalName` sounds indeed like a much better candidate.

Reply:
I found a better method to expose..  luckily I'm having flu so I needed something easy and boring to work on so I could do this already.
Removed the `sanitizeName` method.